src/monitor.ts

```diff
@@ -0,0 +1,357 @@
+/**
+ * This file implements a monitor process for network nodes. It takes a path to
+ * a writeable jsonline file (.jl) as configuration. Each line should define a
+ * json object corresponding to the interface `IMonitoredNodeConfig`. The
+ * monitor process may modify the file and add more objects to it.
+ */
+
+import * as express      from 'express';
+import * as winston      from 'winston';
+import * as program      from 'commander';
+import * as readline     from 'readline';
+import * as fs           from 'fs';
+import * as request      from 'request';
+import * as bodyParser   from 'body-parser';
+import * as ChildProcess from 'child_process';
+
+interface IMonitoredNodeConfig {
+  /**
+   * The rpc address of the monitored node. In IP:port format.
+   */
+  address: string;
+
+  /**
+   * A command that can be used to start/restart the node.  If `undefined`, the
+   * the monitor will not try to restart the node.
+   */
+  reviveCmd?: string;
+
+  /**
+   * The arguments that should be passed to the cmd above, seperated by space.
+   */
+  reviveArgs?: string;
+};
+
+
+/**
+ * Takes any object and checks if it conforms to the config interface.
+ */
+function validateMonitoredNodeConfig(data: any) : IMonitoredNodeConfig | undefined {
```

Comment:
also i wonder

1. if there's a better way (library? or just ajv) to do this since it looks rather repetitive
2. what exactly is the goal of validating the body